app/dao/notifications_dao.py

```diff
@@ -89,11 +90,15 @@ def dao_create_notification(notification):
     db.session.add(notification)
 
 
-def _decide_permanent_temporary_failure(current_status, status):
+def _decide_permanent_temporary_failure(current_status, status, code=None):
     # Firetext will send pending, then send either succes or fail.
-    # If we go from pending to delivered we need to set failure type as temporary-failure
+    # If we go from pending to failure we need to set failure type as temporary-failure
+    # if we get a detailed code from firetext, we should use that code to set status instead
     if current_status == NOTIFICATION_PENDING and status == NOTIFICATION_PERMANENT_FAILURE:
-        status = NOTIFICATION_TEMPORARY_FAILURE
+        if code:
+            status = get_message_status_from_firetext_code(code)
```

Comment:
If we get a code back from firetext which isn't in our dict (maybe they change their codes or add a new code and we aren't aware), we will throw a `KeyError` here. It feels unlikely though, as that doc they sent has changed in 4 years. Does that feel OK? I'm not decided if I think it's fine like that or we should catch it and do something with it (but I don't know what I think we should do).

What do you think?

Reply:
Ahh, good point, I think we should catch it and proceed with our old method as a fallback, and at the same time log an error with the new / unknown code in it, so we can investigate and add it to our list of codes. 

What do you think?

Reply:
Sounds like a good solution, good for it.